packages/dds/merge-tree/src/attributionUtils.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils/internal";
+import type { CustomAttributionKey } from "@fluidframework/runtime-definitions/internal";
+
+import type { PropertySet } from "./properties.js";
+
+/**
+ * @internal
+ */
+export const customAttributionKeysPropName: string =
+	"CAK-cf9b6fe4-4c50-4a5d-9045-eb73b886f740";
+
+/**
+ * @internal
+ */
+export interface ICustomAttributionKeyList {
+	type: "custom";
+	keys: { offset: number; key: CustomAttributionKey }[];
+}
+
+export function isCustomAttributionKeyList(obj: unknown): obj is ICustomAttributionKeyList {
+	return (
+		obj !== null &&
+		typeof obj === "object" &&
+		"type" in obj &&
+		"keys" in obj &&
+		obj.type === "custom" &&
+		Array.isArray(obj.keys)
```

Comment:
I think you can simply that:

```ts
obj?.type === "custom" &&
Array.isArray(obj?.keys)
```

That said, this feels like pretty weak detection of a type system.
Should it has some unique (long) property that is tested here?
I.e. `obj?._customAttributionKeyList === true`